### PR TITLE
Improved ApiAuth middleware

### PR DESCRIPTION
**Pull request by Paplow**
Since ApiAuth middleware is in the "api" middleware group it always compare the value of the header token to the value of api_token in the handesk config file without checking if the token is present in the header. Like this, any request to routes with "api" middleware group will return a 403 Unauthorized status.

So i improved it to make sure it checks to see if the token header is present before comparing it to the handesk api_token value. Allowing for more flexibility to routes in the "api" middleware group.

### CODE REVIEW

File: app/Http/Middleware/ApiAuth.php [hunk 1]
```diff
@@ -10,15 +10,15 @@ class ApiAuth
...
+     * @param  \Illuminate\Http\Request $request
+     * @param  \Closure $next
      *
      * @return mixed
      */
     public function handle($request, Closure $next)
     {
-        if (request()->header('token') != config('handesk.api_token')) {
-            return response()->json(['error' => 'unauthorized'], Response::HTTP_FORBIDDEN);
+        if (request()->header('token') && (request()->header('token') != config('handesk.api_token'))) {
```
Comment (BadChoice) [inline app/Http/Middleware/ApiAuth.php]: I believe this can be unsafe as if the user doesn't send the token header, it won't get the unauthorized response

Comment (BadChoice) [reply app/Http/Middleware/ApiAuth.php]: but won't you be able to use the api if you don't send any token?

Comment (Paplow) [reply app/Http/Middleware/ApiAuth.php]: Yeah.. that's true, Will send another PR that corrects it. I think its better setting ApiAuth middleware in the route group that requires it..

Comment (BadChoice) [reply app/Http/Middleware/ApiAuth.php]: middleware in the route group that corrects it sounds ok to me

Comment (BadChoice) [reply app/Http/Middleware/ApiAuth.php]: Can you fix that no token authorization first so I can merge this one?
Thank you!

**Comment by Paplow** [comment]
Corrected it in my latest commit https://github.com/BadChoice/handesk/pull/158/commits/31aed5d6d0716ba1f0a312b88b63ffb4e3b08e85

File: app/Http/Middleware/ApiAuth.php (hunk 1 above)
Comment (Paplow) [reply app/Http/Middleware/ApiAuth.php]: Corrections has been made in my latest commit https://github.com/BadChoice/handesk/pull/158/commits/31aed5d6d0716ba1f0a312b88b63ffb4e3b08e85

**Comment by BadChoice** [comment]
Thank you so much!
Looks good to me, Just merged it

**Pull request merged**